dio/lib/src/options.dart

```diff
@@ -50,22 +50,22 @@ typedef RequestEncoder = List<int> Function(
 /// `dio.options` is a instance of [BaseOptions]
 class BaseOptions extends _RequestConfig {
   BaseOptions({
-    String method,
+    String? method,
     this.connectTimeout,
-    int receiveTimeout,
-    int sendTimeout,
-    this.baseUrl,
-    this.queryParameters,
-    Map<String, dynamic> extra,
-    Map<String, dynamic> headers,
-    ResponseType responseType = ResponseType.json,
-    String contentType,
-    ValidateStatus validateStatus,
-    bool receiveDataWhenStatusError = true,
-    bool followRedirects = true,
-    int maxRedirects = 5,
-    RequestEncoder requestEncoder,
-    ResponseDecoder responseDecoder,
+    int? receiveTimeout,
+    int? sendTimeout,
+    String? baseUrl,
```

Comment:
It is easier here to just use a default parameter instead of doing the check inside of the constructor.
```suggestion
    this.baseUrl = '',
```
You can then also remove `late` from `baseUrl`.